Tribler/Test/Core/TFTP/test_tftp_handler.py

```diff
@@ -0,0 +1,160 @@
+from nose.tools import raises
+from Tribler.Core.TFTP.exception import FileNotFound
+from Tribler.Core.TFTP.handler import TftpHandler
+from Tribler.Core.TFTP.packet import OPCODE_OACK, OPCODE_ERROR
+from Tribler.Test.Core.base_test import TriblerCoreTest, MockObject
+from Tribler.dispersy.util import blocking_call_on_reactor_thread
+
+
+class TestTFTPHandler(TriblerCoreTest):
+    """
+    This class contains tests for the TFTP handler class.
+    """
+
+    def setUp(self, annotate=True):
+        TriblerCoreTest.setUp(self, annotate=annotate)
+        self.handler = TftpHandler(None, None, None)
+
+    @blocking_call_on_reactor_thread
+    def test_download_file_not_running(self):
+        """
+        Testing whether we do nothing if we are not running a session
+        """
+        def mocked_add_new_session(_):
+            raise RuntimeError("_add_new_session not be called")
+
+        self.handler._add_new_session = mocked_add_new_session
+        self.handler.download_file("test", "127.0.0.1", 1234)
+
+    def test_check_session_timeout(self):
+        """
+        Testing whether we fail if we exceed our maximum amount of retries
+        """
+        mock_session = MockObject()
+        mock_session.retries = 2
+        mock_session.timeout = 1
+        mock_session.last_contact_time = 2
+        self.handler._max_retries = 1
+        self.assertTrue(self.handler._check_session_timeout(mock_session))
+
+    def test_schedule_callback_processing(self):
+        """
+        Testing whether we do not register a tftp process task again
+        """
+        self.handler._callback_scheduled = True
+        self.handler._schedule_callback_processing()
+        self.assertFalse(self.handler.is_pending_task_active("tftp_process_callback"))
```

Comment:
Maybe assert that before it was not pending, but by setting it to true it is (and the call having no effect) ?